.sync/github_templates/contributing/CONTRIBUTING.md

```diff
@@ -46,6 +46,35 @@ If you cannot find an existing issue that describes your bug or feature, create
 Please continue to follow your request after it is submitted to assist with any additional information that might be
 requested.
 
+### Pull Request Best Practices
+
+Generally pull requests for UEFI code can become large and difficult to review due to the large number of build and
+configuration files. To aid maintainers in reviewing your code we suggest adhering to the following guidelines:
+
+1. Do keep code reviews single purpose; don't add more than one feature at a time.
+2. Do fix bugs independently of adding features.
+3. Do provide documentation and unit tests.
+4. Do introduce code in digestable amounts; don't add more than 1000 lines per pull request.
+
+As a general guide to help keep code digestable, you may consider breaking large pull requests into three smaller
+pull requests the the general guide:
+
+1. Interfaces (.h, .inf, documentation)
+2. Implementatation (.c, unit-tests, unit-test build file); unit tests should build and run at this point
+3. Integration/Build (.dec, .dsc, .fdf, integration tests); code added to platform and affects downstream consumers
+
+By breaking the pull request into these three steps the reviewer can digest each piece independently, but each without
+risk of breaking the main branch since without step 3 the code is shipping darkly (not directly integrated into the
+build).
+
+If any of these 3 pull requests is still larger than 1000 lines of code consider breaking the pull request down by the
+following boundaries:
+
+1. By .inf file; break each library/driver into its own 3 part pull request
+2. By .c file; for large drivers break each .c file and its tests into a separate pull request
+
+Feel free to create a draft pull request and ask for suggestions on how to split the pull request if you are unsure.
+

```

Comment:
Content looks great.

Since this is a documentation only change, I'm making some adjustments to improve comprehension speed by reducing some verbosity where possible. Also, some minor grammar modifications as well.

Here's other policy changes I suggest:

1. We do rebase & ff, so for the breakdown of interfaces, implementation, and build, I'd prefer it to be in a single PR so we can collectively see the changes, collectively keep them in sync, and have the same reviewers review all aspects of the change, but have it broken up by commit within the PR as the commits should be shown in git history.

   Within the PR, we can review the commits individually similar to separate PRs. Then, we will simply merge that single PR with those commits as-is.

2. I'd prefer not to put a quantitative limit on the lines of code as it's very circumstantial and I think 1,000 lines will be exceeded very often.

   For me, it is not necessarily the number of lines of code that's a problem, it is the way it is structured. That also impacts maintainability/readability in general.

   So, I'd rather place a "soft limit" on function length and file length as that will provide value beyond the code review and make the review easier. Most important is function length and I'd prefer functions to be contained within one screen full of space - which for me is about 60 lines comfortably. So, I made that 60 lines.

   Super long files often mean poor cohesion which makes review and maintenance more difficult. I propose a "soft limit" file length of around 1,000 lines of code (not including comments).

   By "soft limit", I mean reviewers can reasonably request a refactor due to length alone and it be accepted that's not "nit picking" but a codebase guideline.

3. I changed "3 pull requests" to "3 categories" to abstract the idea strictly from pull requests or commits. The suggestion is to organize it by commits, but we might request by pull request or across two commits in especially convoluted scenarios.

4. I moved `.dec` to "interfaces" instead of "integration/build". The dec declares the package interfaces and these are often needed for the DSC/INF changes in "interfaces".

5. I added section headers to help break content up.

I've included all of my suggestions in the below modification.

```suggestion
### Pull Request Best Practices

Pull requests for UEFI code can become large and difficult to review due to the large number of build and
configuration files. To aid maintainers in reviewing your code, we suggest adhering to the following guidelines:

1. Do keep code reviews single purpose; don't add more than one feature at a time.
2. Do fix bugs independently of adding features.
5. Do provide documentation and unit tests.
7. Do introduce code in digestible amounts.
   - If the contribution logically be broken up into separate pull requests that independently build and function successfully, do use multiple pull requests.

#### Code Categories

To keep code digestible, you may consider breaking large pull requests into three categories of commits within the pull request.

1. **Interfaces**: .h, .inf, .dec, documentation
2. **Implementation**: .c, unit tests, unit test build file; unit tests should build and run at this point
3. **Integration/Build**: .dec, .dsc, .fdf, integration tests; code added to platform and affects downstream consumers

By breaking the pull request into these three categories, the pull request reviewers can digest each piece independently.

If your commits are still very large after adhering to these categories, consider further breaking the pull request down by library/driver; break each component into its own commit.

#### Implementation Limits

Implementation is ultimately composed of functions as logical units of code.

To help maintainers review the code and improve long-term maintainability, limit functions to 60 lines of code. If your function exceeds 60 lines of code, it likely has also exceeded a single responsibility and should be broken up.

Furthermore, files contain functions. Files are easier to review and maintain if they contain functions that serves a similar purpose. Limit files to around 1,000 lines of code (excluding comments). If your file exceeds 1,000 lines of code, it may have functions that should be split into separate files.

---

By following these guidelines, your pull requests will be reviewed faster, and you'll avoid being asked to refactor the code to follow the guidelines.

Feel free to create a draft pull request and ask for suggestions on how to split the pull request if you are unsure.
```